### PR TITLE
Support for element rename and creation of unique names.

### PR DESCRIPTION
**Pull request by niklasharrysson**

### CODE REVIEW

File: source/MaterialXRuntime/Private/PrvElement.h [hunk 1]
```diff
@@ -140,15 +149,13 @@ class PrvElement : public PrvObject
     static const string PATH_SEPARATOR;
 
 protected:
-    PrvElement(RtObjType objType, PrvElement* parent, const RtToken& name);
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/Private/PrvElement.h]: Removed setting parent in element constructors. Instead a child's parent is set in addChild, when the actual child-parent relationship is created.

File: source/MaterialXTest/Runtime.cpp [hunk 2]
```diff
@@ -338,13 +338,26 @@ TEST_CASE("Runtime: Nodes", "[runtime]")
     REQUIRE(add2_in1.getSourcePort() == add1_out);
     REQUIRE(add2_in2.getSourcePort() == add1_out);
 
+    // Test node renaming
```
Comment (feldstj) [inline source/MaterialXTest/Runtime.cpp]: Should probably also test the case where you remove foo1 and try to create foo again.

File: source/MaterialXRuntime/Private/PrvNode.h [hunk 3]
```diff
@@ -84,6 +76,14 @@ class PrvNode : public PrvAllocatingElement
     static void disconnect(const RtPort& source, const RtPort& dest);
 
 protected:
+    // Constructor creating a node with a fixed interface
+    // This is the constructor to use for ordinary nodes.
+    PrvNode(const RtToken& name, const PrvObjectHandle& nodedef, RtObjType objType = RtObjType::NODE);
```
Comment (feldstj) [inline source/MaterialXRuntime/Private/PrvNode.h]: If one of these if for nodes and the other nodegraphs, why even allow other objTypes?

Comment (niklasharrysson) [reply source/MaterialXRuntime/Private/PrvNode.h]: It's a good point. This was an attempt to try and clarify in the code what the intent with the two constructs are, but it's probably just more confusing. So I will remove it. The constructors are protected now anyway and it's the private API so it's only us that will use these constructors.

**Pull request merged**